src/operators_lazysum.jl

```diff
@@ -129,7 +129,7 @@ end
 
 identityoperator(::Type{<:LazySum}, ::Type{S}, b1::Basis, b2::Basis) where S<:Number = LazySum(identityoperator(S, b1, b2))
 
-function embed(basis_l::CompositeBasis, basis_r::CompositeBasis, indices, op::LazySum)
+function embed(basis_l::CompositeBasis, basis_r::CompositeBasis, indices::AbstractVector, op::LazySum)
```

Comment:
I guess this doesn't break any test, but why do we need it?

Reply:
It leads to a method ambiguity if it is not restricted:

```
1 ambiguities found
Ambiguity #1
embed(basis_l::QuantumOpticsBase.CompositeBasis, basis_r::QuantumOpticsBase.CompositeBasis, index::Integer, op::QuantumOpticsBase.AbstractOperator) @ QuantumOpticsBase ~/Documents/ScratchSpace/quantumoptics/QuantumOpticsBase/src/operators.jl:173
embed(basis_l::QuantumOpticsBase.CompositeBasis, basis_r::QuantumOpticsBase.CompositeBasis, indices, op::QuantumOpticsBase.LazySum) @ QuantumOpticsBase ~/Documents/ScratchSpace/quantumoptics/QuantumOpticsBase/src/operators_lazysum.jl:132
```

There are probably other feasible ways to fix this if it is a concern

Reply:
I think we might be able to kill the first method listed instead.

Reply:
Is it enough to provide `embed(basis_l::CompositeBasis, basis_r::CompositeBasis, indices::Integer, op::LazySum)` and `embed(basis_l::CompositeBasis, basis_r::CompositeBasis, indices, op::LazySum)`?